File: Java/DesignPatterns/src/test/java/designPatterns/creational/factoryMethod/MainFactoryMethodTest.java
package designPatterns.creational.factoryMethod;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MainFactoryMethodTest {
  private Creator cre;

  @BeforeEach
  void setUp() {
    cre = new ConcreteCreator();
  }

  @Test
  void equalsTest() {
    assertThat(cre.someOperation()).isEqualTo("Using ConcreteProduct");
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/creational/factoryMethod/MainFactoryMethodTest.kt
package creational.factoryMethod

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class MainFactoryMethodTest {

  private lateinit var cre: Creator

  @BeforeEach
  internal fun setUp() {
    cre = ConcreteCreator()
  }

  @Test
  internal fun equalsTest() {
    assertThat(cre.someOperation()).isEqualTo("Using ConcreteProduct")
  }
}